src/main/java/org/motechproject/newebodac/service/FieldConfigService.java

```diff
@@ -94,4 +100,26 @@ public void delete(UUID id) {
 
     fieldConfigRepository.delete(fieldConfig);
   }
+
+  /**
+   * Checks uniqueness of the field value.
+   * @param entityType type of checked entity.
+   * @param entity checked entity.
+   */
+  public void checkIfUnique(EntityType entityType, BaseEntity entity, UUID id) {
+    List<FieldConfig> fieldConfigs = fieldConfigRepository.findByEntity(entityType);
+    try {
+      Class<?> entityClass = entity.getClass();
+      for (FieldConfig fieldConfig : fieldConfigs) {
+        if (fieldConfig.getUniqueField()) {
+          String fieldName = fieldConfig.getName();
+          Field field = entityClass.getDeclaredField(fieldName);
+          field.setAccessible(true);
+          fieldConfigRepository.checkIfUnique(entityType, entity, id, fieldName, field);
+        }
+      }
+    } catch (NoSuchFieldException e) {
+      LOGGER.error(e.getMessage(), e);
```

Comment:
I think you should throw validation exception here (you can create new exception class for this if it's not exist)